tests/basic/sharedv4_svc_test.go

```diff
@@ -171,11 +177,331 @@ var _ = Describe("{NFSServerFailover}", func() {
+					func() {
+						newRepl, err := Inst().V.GetReplicationFactor(vols[0])
+						Expect(err).NotTo(HaveOccurred())
+						Expect(newRepl).To(BeNumerically("==", 2))
+						// ValidateContext() will fail with error "volume has invalid repl value. Expected:3 Actual:2"
+						// without the line below.
+						ctx.SkipVolumeValidation = true
+					})
+			}
+		})
+
+		Step("for each context, restart volume driver on NFS server node and verify I/O", func() {
+			for _, ctx := range testSv4Contexts {
+				vols, err := Inst().S.GetVolumes(ctx)
+				Expect(err).NotTo(HaveOccurred())
+
+				apiVol, err := Inst().V.InspectVolume(vols[0].ID)
+				Expect(err).NotTo(HaveOccurred())
+
+				// verify failover and failback by repeating the steps below
+				for i := 0; i < 2; i++ {
+					var countersBefore, countersAfter map[string]appCounter
+					var attachedNodeBefore, attachedNodeAfter *node.Node
+
+					// get the attached node before the failover
+					attachedNodeBefore, err = Inst().V.GetNodeForVolume(vols[0],
+						defaultCommandTimeout, defaultCommandRetry)
+					Expect(err).NotTo(HaveOccurred())
+
+					Step(
+						fmt.Sprintf("get counter values from node %v for app %s's volume before failover",
+							attachedNodeBefore.Name, ctx.App.Key),
+						func() {
+							countersBefore = getAppCounters(apiVol, attachedNodeBefore,
+								3*time.Duration(numPods)*time.Second)
+						})
+
+					Step(
+						fmt.Sprintf("failover #%d for app %s by stopping the volume driver on node %s",
+							i, ctx.App.Key, attachedNodeBefore.Name),
+						func() {
+							StopVolDriverAndWait([]node.Node{*attachedNodeBefore})
+						})
+
+					Step(
+						fmt.Sprintf("start volume driver %s on node where app %s's volume was attached: %s",
+							Inst().V.String(), ctx.App.Key, attachedNodeBefore.Name),
+						func() {
+							StartVolDriverAndWait([]node.Node{*attachedNodeBefore})
+						})
+
+					Step("Giving few seconds for volume driver and the app pods to stabilize",
+						func() {
+							time.Sleep(60 * time.Second)
+						})
+
+					Step(fmt.Sprintf("validate app %s", ctx.App.Key),
+						func() {
+							ValidateContext(ctx)
+						})
+
+					Step(
+						fmt.Sprintf("get counter values for app %s's volume after failover #%d", ctx.App.Key, 0),
+						func() {
+							attachedNodeAfter, err = Inst().V.GetNodeForVolume(vols[0],
+								defaultCommandTimeout, defaultCommandRetry)
+							Expect(err).NotTo(HaveOccurred())
+							Expect(attachedNodeAfter.Name).NotTo(Equal(attachedNodeBefore.Name))
+							countersAfter = getAppCounters(apiVol, attachedNodeAfter,
+								3*time.Duration(numPods)*time.Second)
+						})
+
+					Step(fmt.Sprintf("validate no I/O disruption for app %s after failover #%d", ctx.App.Key, i),
+						func() {
+							// 2 pods (1 on the old NFS server and 1 on the new NFS server) should get restarted,
+							// others should not be interrupted
+							validateAppCounters(ctx, countersBefore, countersAfter, numPods, 2)
```

Comment:
Should we do an explicit check here that only the pods which were running on the old and the new server got restarted?

Reply:
The existing test in this file verfies that already. See line 140 above:
```
// make sure the pods on both old and new server are restarted
```